js/components/generate/__tests__/generate.test.js

```diff
@@ -0,0 +1,129 @@
+import React from "react"
+import GenerateForm from "../GenerateForm.js"
+import { screen, render, fireEvent, waitFor, within } from "@testing-library/react"
+import courseToResp from "../__mocks__/sample_responses.json"
+
+const origFetch = global.fetch
+
+const mocker = courseInput => {
```

Comment:
Having the `sample_responses.json` file is good. However, you can improve on the mocking structure by using `fetchMock` in `setupTests.js`, like the existing mocking. Note that the "response" argument to `fetchMock.mock`/`fetchMock.get`/`fetchMock.put`/etc. can be a *function* that takes information about the request and then determines what the response should be. So, you can keep the json file the way you have it, but use it in a single `fetchMock.put` call and dynamically determine what to return. See https://www.wheresrhys.co.uk/fetch-mock/#api-mockingmock_response.